Fixing build broken links for netlify build (remove from todoPatterns)

jaiakash opened the pull request:
Hi, I removed all (for now yes, all) links from `todoPatterns` to check the status of netlify build.

Update 1: When i removed all the links total 6 tests were failing. 
<img width="1240" alt="Screenshot 2025-01-18 at 12 42 51 PM" src="https://github.com/user-attachments/assets/baad7465-e6ff-4f68-b264-ffc47d747c5a" />

After reverting those link patterns, build is passing. 

Fixes #3960

Arhell reviewed (commented):
/ok-to-test

hbelmiro reviewed (commented):
@jaiakash Have you verified where the pending links are? Please consider using `skipPatterns` as mentioned by Mathew in the issue.

> NOTE: if some of the TODO links are actually correct but the plugin is making a mistake, we can use the skipPatterns section instead, so they don't appear in the logs.

jaiakash commented:
Hi @hbelmiro, So I tried to dig deeper into the issue. There are 4 patterns that are causing issues right now. I will go through one by one.

1. `public/docs/started/getting-started`
This is the legacy page for getting started/installation for KubeFlow. Currently, 2 places ( first is [here](https://github.com/kubeflow/website/blob/f82061c18d1349fd972da7bf59675314645fea65/content/en/docs/components/training/user-guides/mpi.md?plain=1#L31C51-L31C105)  and 2nd is [here](https://github.com/kubeflow/website/blob/f82061c18d1349fd972da7bf59675314645fea65/content/en/docs/started/support.md?plain=1#L115) ) where we are using this link.

**Solution:**  We can replace it with KubeFlow installation link.

2. `public/docs/distributions/gke/`
I think these pages are permanently moved or deleted for the current version of docs. There is no occurrence of this pattern in new docs as relative, only `legacy-v1` has these links as relative path. New docs uses absolute path for GKE. Check below line
https://github.com/kubeflow/website/blob/f82061c18d1349fd972da7bf59675314645fea65/content/en/docs/components/pipelines/concepts/pipeline-root.md?plain=1#L30

**Solution:** We have to use GKE docs absolute link for referencing here https://googlecloudplatform.github.io/kubeflow-gke-docs/docs

jaiakash commented:
3. `public/docs/other-guides` 
I did not understand what this other guide meant in docs. The link itself is broken on the production version of docs. Click on **Further setup and troubleshooting** on this page https://www.kubeflow.org/docs/started/support/#support-from-the-kubeflow-community
Let me know what this link is supposed to do.

4. `public/docs/examples/shared-resources`
This pattern exists only in `legacy-v1` docs only. I think this link to supposes to redirect to the examples page. 

**Solution:**  Replace this link with examples page (https://www.kubeflow.org/docs/started/kubeflow-examples/)

jaiakash commented:
Hi @hbelmiro @thesuperzapper, can you please review and give your suggestions?

hbelmiro commented:
@jaiakash for [3], I suggest removing that bullet.
All the other solutions proposed sound good to me.

jaiakash commented:
Hi @hbelmiro , 

> 1. `public/docs/started/getting-started`
>    This is the legacy page for getting started/installation for KubeFlow. Currently, 2 places ( first is [here](https://github.com/kubeflow/website/blob/f82061c18d1349fd972da7bf59675314645fea65/content/en/docs/components/training/user-guides/mpi.md?plain=1#L31C51-L31C105)  and 2nd is [here](https://github.com/kubeflow/website/blob/f82061c18d1349fd972da7bf59675314645fea65/content/en/docs/started/support.md?plain=1#L115) ) we are using this link.
> 
> **Solution:** We can replace it with KubeFlow installation link.
> 
> 2. `public/docs/distributions/gke/`
>    I think these pages are permanently moved or deleted for the current version of docs. There is no occurrence of this pattern in new docs as relative path, only `legacy-v1` has these links as relative path. New docs uses absolute path for GKE. 
> **Solution:** We have to use GKE docs absolute link for referencing above pattern. [GKE Docs](https://googlecloudplatform.github.io/kubeflow-gke-docs/docs)

For 1, I changed the link to KubeFlow installation docs.
For 2, I tried using Kubeflow link and then redirecting to GKE docs, but still the tests were failing. So I changed to actual GKE docs links.



> 3. `public/docs/other-guides`
>    I did not understand what this other guide meant in docs. The link itself is broken on the production webssite. Click on **Further setup and troubleshooting** on this page [kubeflow.org/docs/started/support#support-from-the-kubeflow-community](https://www.kubeflow.org/docs/started/support/#support-from-the-kubeflow-community)
>    Let me know what this link is supposed to do.
> 4. `public/docs/examples/shared-resources`
>    This pattern exists only in `legacy-v1` docs only. I think this link to supposed to redirect to the examples page.
> 
> **Solution:** Replace this link with examples page ([kubeflow.org/docs/started/kubeflow-examples](https://www.kubeflow.org/docs/started/kubeflow-examples/))

For 3, I deleted it since its not needed now.

For 4, shared resources are supposed to point at examples but not for all the pages, so I moved it to skipPatterns

hbelmiro reviewed (approved):
/lgtm

jaiakash commented:
Any update on PR @hbelmiro ?

hbelmiro commented:
@andreyvelich @terrytangyuan can one of you guys PTAL?

terrytangyuan reviewed (approved):
Thanks!

/lgtm
/approve
